doc/adr/0003-use_pundit_for_rbac.md

```diff
@@ -0,0 +1,57 @@
+# 3. Use Pundit for RBAC, Store Roles in Cognito and use Cognito Group for teams
+
+Date: 2019-07-30
+
+## Status
+
+Pending
+
+## Context
+
+As with any modern web system there is a need to secure the application with
+stong authentication and Role Based Access Control (RBAC). RBAC allows us to
+create permissions, apply them to roles and and associates roles with users.
+From this we can restrict what users can do and see with in the application
+based ontheir role. In addition to this we also had to investiage where to store
+role information and define teams to support our use of RBAC.
+
+## Decision
+
+After some investigation and discussion in to various ways of implementing RBAC
+within ruby we have decided to implement the Pundit Ruby gem within the verify
+self service application. This gem provide us with away of creating policies
+which applies to the app as a whole and to individual classes within the
+application.
+
+Whave chosen pundit for a number of reasons. The first is that it seems
+relatively easy to implment within our application. A proof of concept was
+created in a few hours which restricted creating new components within the
```

Comment:
could maybe put in a gist/on a branch and link to proof of concept?

Reply:
We could it ... but it would take an hours work or so to unpick it a bit